### DISCUSSION TITLE
Allow ObservableProperty to be used in classes implementing INotifyPropertyChanged

### DISCUSSION THREAD

**Discussion by Moreno-Gentili** [CommunityToolkit/Maui#1129] [New Feature Discussions]
Hi everyone, suppose I liked my view models to be close to their view. So close, in fact, that I'd define my commands and properties in the code behind file (easier to understand, less files to manage).

```csharp
[ObservableObject]
public partial class MyClass : ContentView
{
    public MyClass()
    {
        InitializeComponent();
        BindingContext = this;
    }

    [ObservableProperty]
    int myProperty;

    [RelayCommand]
    void MyCommand()
    {
        MyProperty += 1;
    }
}
```

Unfortunately, this is not allowed. The error [MVVMTK0002](https://learn.microsoft.com/it-it/dotnet/communitytoolkit/mvvm/generators/errors/mvvmtk0002) is shown.
```
Cannot apply `[ObservableObject]` to type {type}, as it already declares the `INotifyPropertyChanged` interface.
```

What's exactly the problem here? Since a `ContentView` is already implementing `INotifyPropertyChanged`, the source generator should just add the missing parts.

**Comment by TheCodeTraveler** [comment]
Hi @Moreno-Gentili!

You'll want to open a discussion on the MVVM Toolkit repo, https://GitHub.com/CommunityToolkit/dotnet

**Comment by Moreno-Gentili** [comment]
Thanks, I'm closing this.
